vllm/entrypoints/openai/serving_embedding.py

```diff
@@ -88,16 +89,11 @@ async def create_embedding(
         request_id = f"embd-{self._base_request_id(raw_request)}"
         created_time = int(time.time())
 
-        truncate_prompt_tokens = None
+        truncate_prompt_tokens = request.truncate_prompt_tokens
 
-        if request.truncate_prompt_tokens is not None:
-            if request.truncate_prompt_tokens <= self.max_model_len:
-                truncate_prompt_tokens = request.truncate_prompt_tokens
-            else:
-                return self.create_error_response(
-                    "truncate_prompt_tokens value is "
-                    "greater than max_model_len."
-                    " Please, select a smaller truncation size.")
+        if truncate_prompt_tokens is not None:
```

Comment:
You can move this `if` into the `_validate_truncation_size()` function to save some lines of code.